pkg/descheduler/strategy_migration.go

```diff
@@ -166,106 +142,214 @@ var pluginsMap = map[string]func(ctx context.Context, nodes []*v1.Node, params *
-			return
-		}
-		status := pg.(framework.BalancePlugin).Balance(ctx, nodes)
-		if status != nil && status.Err != nil {
-			klog.V(1).ErrorS(err, "plugin finished with error", "pluginName", removeduplicates.PluginName)
+			klog.ErrorS(err, "unable to validate plugin arguments", "pluginName", removeduplicates.PluginName)
+			return nil, fmt.Errorf("strategy %q param validation failed: %v", removeduplicates.PluginName, err)
 		}
+		return &api.PluginConfig{
+			Name: removeduplicates.PluginName,
+			Args: args,
+		}, nil
 	},
-	"RemovePodsViolatingTopologySpreadConstraint": func(ctx context.Context, nodes []*v1.Node, params *api.StrategyParameters, handle *handleImpl) {
+	"RemovePodsViolatingTopologySpreadConstraint": func(params *v1alpha1.StrategyParameters) (*api.PluginConfig, error) {
 		args := &removepodsviolatingtopologyspreadconstraint.RemovePodsViolatingTopologySpreadConstraintArgs{
-			Namespaces:             params.Namespaces,
+			Namespaces:             v1alpha1NamespacesToInternal(params.Namespaces),
 			LabelSelector:          params.LabelSelector,
 			IncludeSoftConstraints: params.IncludeSoftConstraints,
 		}
 		if err := removepodsviolatingtopologyspreadconstraint.ValidateRemovePodsViolatingTopologySpreadConstraintArgs(args); err != nil {
-			klog.V(1).ErrorS(err, "unable to validate plugin arguments", "pluginName", removepodsviolatingtopologyspreadconstraint.PluginName)
-			return
-		}
-		pg, err := removepodsviolatingtopologyspreadconstraint.New(args, handle)
-		if err != nil {
-			klog.V(1).ErrorS(err, "unable to initialize a plugin", "pluginName", removepodsviolatingtopologyspreadconstraint.PluginName)
-			return
-		}
-		status := pg.(framework.BalancePlugin).Balance(ctx, nodes)
-		if status != nil && status.Err != nil {
-			klog.V(1).ErrorS(err, "plugin finished with error", "pluginName", removepodsviolatingtopologyspreadconstraint.PluginName)
+			klog.ErrorS(err, "unable to validate plugin arguments", "pluginName", removepodsviolatingtopologyspreadconstraint.PluginName)
+			return nil, fmt.Errorf("strategy %q param validation failed: %v", removepodsviolatingtopologyspreadconstraint.PluginName, err)
 		}
+		return &api.PluginConfig{
+			Name: removepodsviolatingtopologyspreadconstraint.PluginName,
+			Args: args,
+		}, nil
 	},
-	"HighNodeUtilization": func(ctx context.Context, nodes []*v1.Node, params *api.StrategyParameters, handle *handleImpl) {
+	"HighNodeUtilization": func(params *v1alpha1.StrategyParameters) (*api.PluginConfig, error) {
+		if params.NodeResourceUtilizationThresholds == nil {
+			params.NodeResourceUtilizationThresholds = &v1alpha1.NodeResourceUtilizationThresholds{}
+		}
 		args := &nodeutilization.HighNodeUtilizationArgs{
-			Thresholds:    params.NodeResourceUtilizationThresholds.Thresholds,
-			NumberOfNodes: params.NodeResourceUtilizationThresholds.NumberOfNodes,
+			EvictableNamespaces: v1alpha1NamespacesToInternal(params.Namespaces),
```

Comment:
I have added this way for people to already use preEviction namespace filtering for node utilization even before v1alpha2 is totally here.

Reply:
It is relevant in this PR because without this we don't get the conversion that we expect on the other side with all params being translated.

Reply:
Always good to write tests and notice these things :point_right: https://github.com/kubernetes-sigs/descheduler/pull/999/files#r1020344073